src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,267 @@
+    cbsd_fad_records_iteration_0_sas_test_harness_1 = generateCbsdRecords([sas_test_harness_device_4,sas_test_harness_device_5],
+                                                                          [[grant_request_2, grant_request_3], [grant_request_4]])
+
+    # Generate Cbsd FAD Records for SAS Test Harness 0, iteration 1
+    cbsd_fad_records_iteration_1_sas_test_harness_0 = generateCbsdRecords([sas_test_harness_device_2,sas_test_harness_device_3],
+                                                                          [[grant_request_1],[grant_request_2]])
+
+    # Generate Cbsd FAD Records for SAS Test Harness 1, iteration 1
+    cbsd_fad_records_iteration_1_sas_test_harness_1 = generateCbsdRecords([sas_test_harness_device_6], [[grant_request_5, grant_request_6]])
+   
+    # SAS Test Harnesses configuration  
+    sas_test_harness_0_config = {
+        'sasTestHarnessName': 'SAS-TH-1',
+        'hostName': 'localhost',
+        'port': 9001,
+        'serverCert': os.path.join('certs', 'server.cert'),
+        'serverKey': os.path.join('certs', 'server.key'),
+        'caCert': os.path.join('certs', 'ca.cert')
+    }
+    sas_test_harness_1_config = {
+        'sasTestHarnessName': 'SAS-TH-2',
+        'hostName': 'localhost',
+        'port': 9002,
+        'serverCert': os.path.join('certs', 'server.cert'),
+        'serverKey': os.path.join('certs', 'server.key'),
+        'caCert': os.path.join('certs', 'ca.cert')
+    }
+
+    # Generate SAS Test Harnesses dump records for multiple iterations   
+    sas_test_harness_0_dump_records_iteration_0 = {
+        'cbsdRecords': cbsd_fad_records_iteration_0_sas_test_harness_0
+    }
+    sas_test_harness_0_dump_records_iteration_1 = {
+        'cbsdRecords': cbsd_fad_records_iteration_1_sas_test_harness_0
+    }
+    sas_test_harness_1_dump_records_iteration_0 = {
+        'cbsdRecords': cbsd_fad_records_iteration_0_sas_test_harness_1
+    }
+    sas_test_harness_1_dump_records_iteration_1 = {
+        'cbsdRecords': cbsd_fad_records_iteration_1_sas_test_harness_1
+    }
+
+    # Create the actual config.
+    iteration1_config = {
+        'cbsdRequests': [cbsd_records_iteration_1_domain_proxy_0,cbsd_records_iteration_1_domain_proxy_1],
+        'protectedEntities': protected_entities_iteration_1,
+        'dpaActivationList': [dpa_1, dpa_2],
+        'dpaDeactivationList': [],
+        'sasTestHarnessData': [sas_test_harness_0_dump_records_iteration_0,sas_test_harness_1_dump_records_iteration_0]
+    }
+    iteration2_config = {
+        'cbsdRequests': [cbsd_records_iteration_2_domain_proxy_0,cbsd_records_iteration_2_domain_proxy_1],
+        'protectedEntities': protected_entities_iteration_2,
+        'dpaActivationList': [dpa_3],
+        'dpaDeactivationList': [dpa_1],
+        'sasTestHarnessData': [sas_test_harness_0_dump_records_iteration_1,sas_test_harness_1_dump_records_iteration_1]
+    }
+    config = {
```

Comment:
I just realized that there's another piece missing: we need the capability for individual CBSDs (which will use their own cert/key pair) to register, get a grant, etc. So in the same way we support multiple DPs, we need to support multiple individual CBSD connections.

Sorry for catching this only at the last minute -- I literally remembered only as I was writing a reply to an unrelated email.

Reply:
We are working on this in next drop. Will update accordingly.

Reply:
I believe that the intention is to use individual CBSDs in every step where "DP test harness" is mentioned: so ITS can use N DPs and M individual CBSDs in each of these steps. The test spec is unfortunately not very clear on this point; you are welcome to double-check with Masoud.